### PR TITLE
Major dependencies upgrade

### PR DESCRIPTION
**Pull request by wendigo**
No release notes needed

### CODE REVIEW

**Review by linzebing — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -499,6 +505,18 @@
                 <version>${dep.errorprone.version}</version>
             </dependency>
 
+            <dependency>
+                <groupId>com.google.http-client</groupId>
+                <artifactId>google-http-client</artifactId>
+                <version>1.43.3</version>
```
Comment (electrum) [inline pom.xml:511]: Add a property for this?

File: core/trino-main/pom.xml [hunk 2]
```diff
@@ -473,7 +473,7 @@
         <dependency>
             <groupId>org.assertj</groupId>
             <artifactId>assertj-guava</artifactId>
-            <version>3.4.0</version>
+            <version>${dep.assertj-core.version}</version>
```
Comment (electrum) [inline core/trino-main/pom.xml:476]: We could start using `assertj-bom`

File: pom.xml [hunk 3]
```diff
@@ -2029,6 +2029,37 @@
                     </configuration>
                 </plugin>
 
+                <plugin>
+                    <groupId>org.codehaus.mojo</groupId>
+                    <artifactId>versions-maven-plugin</artifactId>
+                    <version>2.16.1</version>
+                    <configuration>
+                        <ruleSet>
+                            <ignoreVersion>
```
[the remark this answers is not in the document]
Comment (wendigo) [reply pom.xml:2058]: It helps me track dependencies to upgrade. I'll extract it to separate branch

File: core/trino-main/pom.xml (hunk 2 above)
Comment (wendigo) [reply core/trino-main/pom.xml:476]: https://github.com/airlift/airbase/pull/383

**Pull request merged**